src/common/access_log.ts

```diff
@@ -0,0 +1,43 @@
+import {Response, Request, NextFunction} from 'express';
+import {logging} from './logging';
+
+export const accessLogMiddleware = () => {
+  const logger = logging.child({type: 'http'});
+
+  return (req:Request, res:Response, next:NextFunction) => {
+    const startTime = new Date();
+
+    req.on('end', () => {
+      const responseTime:number = new Date().getTime() - startTime.getTime();
+      const statusCode:number = res.statusCode;
+      const details:object = {
+        response_time: responseTime,
+        req: getRequestDetails(req),
+        res: getResponseDetails(res),
+      };
+
+      if (statusCode >= 500) {
```

Comment:
A 5XX HTTP status codes mean the request error because of an error in the server/system. Where a 4XX HTTP status is an error made by the client. 

I'm going to add https://www.npmjs.com/package/http-status-codes